arbnode/tx_pre_checker.go

```diff
@@ -121,15 +123,22 @@ func (c *TxPreChecker) PublishTransaction(ctx context.Context, tx *types.Transac
 	state := c.getLatestState()
 	sender, err := types.Sender(types.LatestSigner(c.bc.Config()), tx)
 	if err != nil {
-		return core.ErrInvalidSender
+		return err
```

Comment:
Not sure why this was changed? Seems like `types.Sender` could return a lot of different error messages

Reply:
This now matches how we reject txs normally in the state transition: In `func (*Transaction) AsMessage`:
```go
	var err error
	msg.from, err = Sender(s, tx)
	return msg, err
```
and in its callsite in `applyTransaction`:
```go
	msg, err := tx.AsMessage(types.MakeSigner(config, header.Number), header.BaseFee)
	if err != nil {
		return nil, err
	}
```

Reply:
I've switched this back, along with the other errors